Lib/asyncio/base_events.py

```diff
@@ -560,7 +565,13 @@ def _asyncgen_firstiter_hook(self, agen):
         try:
             await future
         finally:
-            thread.join()
+            thread.join(timeout)
+
+        if (thread.is_alive()):
+            warnings.warn("The ThreadPoolExecutor did not finishing joining"
+                             f"its threads within {timeout} seconds.",
+                             RuntimeWarning)
```

Comment:
@1st1 Couple of questions regarding the warning:

1) Should it remain a `RuntimeWarning`? I think this would be suitable since it occurs during runtime as a result of the thread not being able to close within the specified timeout window. I could imagine an argument for it being a `ResourceWarning` as well though.

2) Should I add a test to `test_asyncio/test_events.py` to make sure the the warning is triggered in an appropriate scenario? I have some familiarity with the existing tests, but it would be my first time writing a new one.